examples/common/imgui_ui/windows/BUILD.gn

```diff
@@ -51,5 +51,9 @@ static_library("light") {
     "${chip_root}/third_party/imgui",
   ]
 
+  # TODO: this is because on-off-server.h is generally only available
+  #       in the app and we do not want to directly bind to lighting-app
+  check_includes = false
```

Comment:
Just wondering if you have a plan/idea how to deal with that. 

We had a similar issues in a silabs common source set that required the data_model source set as deps. We end up setting a build argument that specify the data model source_set to be used per example but was wondering if there were other options

Reply:
Ideally I believe each server should become stand-alone and not depend on generated files, in which case `light_ui` would be able to depend on `light_server` without pulling in generated files.

However this is probably non-trivial to do today as many implementations require generated data information like ENDPOINT_COUNT. we may need to incrementally decouple things whenever possible.

Unfortunately this is not a plan - it looks like this level of decoupling involves major changes with the only gain being maintenance rather than feature support, so it will not be easy to get resources for such work.